File: Data Structures & Algorithms/combination-target-sum/submission-1.java
class Solution {
    List<List<Integer>> res;
    public List<List<Integer>> combinationSum(int[] nums, int target) {
        res = new ArrayList<>();
        backtrack(nums, target, new ArrayList<>(), 0);
        return res;
    }

    private void backtrack(int[] nums, int target, List<Integer> curr, int j) {
        if (target == 0) {
            res.add(new ArrayList<>(curr));
            return;
        }
        if (target < 0) {
            return;
        }
        for (int i = j; i < nums.length; i++) {
            curr.add(nums[i]);
            backtrack(nums, target - nums[i], curr, i);
            curr.remove(curr.size() - 1);
        }
    }

}

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-1.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l = 0, res = 0;
        for (int r = 0; r < nums.size(); ++r) {
            k -= (nums[r] == 0 ? 1 : 0);
            while (k < 0) {
                k += (nums[l] == 0 ? 1 : 0);
                ++l;
            }
            res = max(res, r - l + 1);
        }
        return res;
    }
};